src/hooks/lib/preview/compounder/useTokenPreviewRedeemFallback.ts

```diff
@@ -0,0 +1,142 @@
+import { useEffect, useState } from "react"
+import { Address, useContractRead } from "wagmi"
+
+import { getGlpPrice } from "@/lib/api/pricer/getGlpPrice"
+import { useActiveChainId } from "@/hooks/useActiveChainId"
+import { useTokenVaultSymbol } from "@/hooks/useTokenVaultSymbol"
+
+import { AMMCompounderBase, GlpVault } from "@/constant/abi"
+import { ETH, glpVault } from "@/constant/addresses"
+
+const WETH_ARBI: Address = "0x82aF49447D8a07e3bd95BD0d56f35241523fBab1"
+
+export default function useTokenPreviewRedeemFallback({
+  asset,
+  vaultAddress,
+  token,
+  amount,
+  slippage,
+  enabled,
+}: {
+  asset: Address
+  vaultAddress: Address
+  token?: Address
+  amount: string
+  slippage: number
+  enabled?: boolean
+}) {
+  const tokenVaultSymbol = useTokenVaultSymbol({
+    asset,
+    enabled,
+  })
+
+  const ybTokenSymbol = tokenVaultSymbol.data
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  let abi: any = undefined
+  if (ybTokenSymbol === "fcGLP") {
+    abi = AMMCompounderBase
```

Comment:
There's only one condition here. I think you can just pass the abi to the contract call as normal.

Reply:
Still would like to see this updated